rust/theoros/src/rpc/mod.rs

```diff
@@ -0,0 +1,60 @@
+pub mod hyperlane;
+pub mod pragma_wrapper;
+
+pub use hyperlane::*;
+pub use pragma_wrapper::*;
+
+use pragma_utils::conversions::starknet::felt_vec_to_vec_string;
+use starknet::{
+    core::types::{BlockId, BlockTag, Felt, FunctionCall},
+    macros::selector,
+    providers::{jsonrpc::HttpTransport, JsonRpcClient, Provider},
+};
+use url::Url;
+
+pub struct StarknetRpc(JsonRpcClient<HttpTransport>);
+
+impl StarknetRpc {
+    pub fn new(rpc_url: Url) -> Self {
+        Self(JsonRpcClient::new(HttpTransport::new(rpc_url)))
+    }
+}
+
+#[async_trait::async_trait]
+impl HyperlaneCalls for StarknetRpc {
+    async fn get_announced_storage_locations(
```

Comment:
we should probably use the rust bindings defined [here](https://github.com/astraly-labs/hyperlane_starknet/blob/main/rust/build.rs) for safety

Reply:
Yup I got an issue for that - will do it in a follow-up 🙏